src/common.c

```diff
@@ -1019,12 +1024,6 @@ int fi_epoll_wait(struct fi_epoll *ep, void **contexts, int max_contexts,
 		if (ep->fds[0].revents)
 			fd_signal_reset(&ep->signal);
 
-		fastlock_acquire(&ep->lock);
-		if (!slist_empty(&ep->work_item_list))
-			fi_epoll_process_work_item_list(ep);
-
-		fastlock_release(&ep->lock);
```

Comment:
We need to keep this within the loop, or we'll ignore updates made after the thread calls poll.

Reply:
agree, I worked in single-thread app. returned this code back to the loop.